dask/array/linalg.py

```diff
@@ -0,0 +1,125 @@
+"""
+   Copyright (c) 2015, Mariano Tepper, Duke University.
+   All rights reserved.
+
+   This file is part of dask and is under the BSD 3-Clause License,
+   which can be found at http://opensource.org/licenses/BSD-3-Clause
+"""
+
+import numpy as np
+from itertools import count, product
+import dask.array as da
+from math import ceil
+import operator
+
+names = ('tsqr_%d' % i for i in count(1))
+
+
+def _findnumblocks(shape, blockshape):
+    def div_ceil(t):
+        return int(ceil(float(t[0]) / t[1]))
+
+    nb = [div_ceil(elem) for elem in zip(*[shape, blockshape])]
+    return tuple(nb)
+
+
+def qr(data, name=None):
```

Comment:
Is it likely that there might be a few different implementations?  If so then we should probably call this function `tsqr` and have a separate `qr` function that looks at the array and determines the right algorithm based on blockdims.